services/construction.go

```diff
@@ -152,7 +169,13 @@ func (s *constructionAPIService) ConstructionHash(
 	ctx context.Context,
 	request *types.ConstructionHashRequest,
 ) (*types.TransactionIdentifierResponse, *types.Error) {
-	terr := ValidateNetworkIdentifier(ctx, s.oasisClient, request.NetworkIdentifier)
+	var terr *types.Error
+	if s.oasisClient == nil {
+		// Running in offline mode, get chain ID from env var.
+		terr = ValidateNetworkIdentifierWithChainID(os.Getenv(OfflineModeChainIDEnvVar), request.NetworkIdentifier)
+	} else {
+		terr = ValidateNetworkIdentifier(ctx, *s.oasisClient, request.NetworkIdentifier)
+	}
```

Comment:
This seems like it is done over and over again, can you move it to a function? E.g., maybe `ValidateNetworkIdentifier` could handle both cases?

Reply:
Good idea, I've changed the `ValidateNetworkIdentifier` function to handle both.